test/hermes_test/model/types/test_ld_container.py

```diff
@@ -67,22 +69,38 @@ def test_container_full_context_and_path(self, httpserver):
         httpserver.expect_request("/url3").respond_with_json({"ham": "bacon"})
         httpserver.expect_request("/url4").respond_with_json({"@context": {"id": "@id"}})
 
-        cont_grand_parent = ld_container({"ham": [{"@value": "eggs"}]}, context=[self.url], index=1)
-        cont_parent = ld_container({"ham": [{"@value": "eggs"}]}, context=[httpserver.url_for("/url2"),
-                                                                           httpserver.url_for("/url4")],
-                                   parent=cont_grand_parent)
-        cont = ld_container({"spam": [{"@value": "bacon"}]}, context=[httpserver.url_for("/url3")], parent=cont_parent,
-                            index=3)
+        cont_data = [{"spam": [{"@value": "bacon"}]}]
+        cont_parent_data = [cont_data]
+        cont_grand_parent = ld_container([{"ham": cont_parent_data}], context=[self.url])
+        cont_parent = ld_container(cont_parent_data, context=[httpserver.url_for("/url2"),
+                                                              httpserver.url_for("/url4")],
+                                   parent=cont_grand_parent, key="ham")
+        cont = ld_container(cont_data, context=[httpserver.url_for("/url3")], parent=cont_parent,
+                            index=0)
         assert cont_parent.full_context == [self.url, httpserver.url_for("/url2"), httpserver.url_for("/url4")]
         assert cont.full_context == [self.url, httpserver.url_for("/url2"), httpserver.url_for("/url4"),
                                      httpserver.url_for("/url3")]
-        assert cont_grand_parent.path == ["$"]          # FIXME: #381 Decision if the path should look like this
-        assert cont_parent.path == ["$", None]
-        assert cont.path == ["$", None, 3]
+        assert cont_grand_parent.path == ["$"]
+        assert cont_parent.path == ["$", "ham"]
+        assert cont.path == ["$", "ham", 0]
 
     def test_container_str_and_repr(self):
-        cont = ld_container({"spam": [{"@value": "bacon"}]}, index=3)
-        assert repr(cont) == ld_container({"spam": [{"@value": "bacon"}]}) #FIXME: #397 KeyError
+        cont = ld_container([{"spam": [{"@value": "bacon"}]}])
+        assert repr(cont) == "ld_container({'spam': [{'@value': 'bacon'}]})"
         with pytest.raises(NotImplementedError):
             str(cont)
 
+    def test_to_python(self):
+        cont = ld_container([{"spam": [{"@value": "bacon", "@id": "ham", "@type": ["@id"]}]}])
+        assert cont._to_python("spam", [{"@value": "bacon"}]) == 'bacon'
+        assert cont._to_python("@id", "ham") == "ham"
+
+        cont.active_ctx['_uuid'] = str(uuid.uuid1())    # FIXME: 406
+        assert cont._to_python("@type", ["@id"]) == '@id'
+
+    def test_to_expanded(self):
+        cont = ld_container([{"spam": [{"@value": "bacon", "@id": "ham", "@type": "@id"}]}])
+        cont.active_ctx['_uuid'] = str(uuid.uuid1())    # FIXME: 406
```

Comment:
Okay, `active_ctx` is something internal to `pyld` and is used here ... as some kind of shortcut. Problem is, that is even for `pyld` something internal and hence not (very well) documented.

I still see that #406 is an issue, though. Again, a stacktrace would be helpful (but I can also run the tests myself for further analysis).

The other question is, how `_to_expanded_json` should behave on undefined properties...